### ISSUE TITLE
kubelet periodic node status update should not be aligned

### ISSUE THREAD

**Issue by mengqiy** [kubernetes/kubernetes#124202]
### What happened?

The default `nodeStatusUpdateFrequency` is 10s with 4% jitter.
The default `nodeStatusReportFrequency` is 5m.

The node status will be updated either there's a change or it has been longer than `nodeStatusReportFrequency` since the previous update.
`nodeStatusReportFrequency` 
https://github.com/kubernetes/kubernetes/blob/55b83c92b3b69cd53d5bf22b8ccff859a005241a/pkg/kubelet/kubelet_node_status.go#L580-L581

The jitter of `nodeStatusUpdateFrequency` won't effectively make the periodic node status update to be spread evenly over the 5-minute window.
The node status update requests from thousands of nodes are still sent at almost the same time after a couple of days. It can cause the CPU usage to spike every 5 minutes.

The following is the APIServer CPU utilization on 96CPU instances. The cluster has 5k nodes that are added in a short period of time. The cluster is idle without any application workload. The utilization can go up to 64% and go down to 14% just because of these  node status update requests.
<img width="1756" alt="Screenshot 2024-04-05 at 12 59 16 PM" src="https://github.com/kubernetes/kubernetes/assets/2589105/cb9eac79-1c12-4baa-84d9-0c7c639d4d40">

### What did you expect to happen?

I'd expect the node status update requests from kubelet to be spread evenly so that the apiserver won't have a minute that it has to handle the requests from most nodes every 5 minutes. 

### How can we reproduce it (as minimally and precisely as possible)?

Bring up a cluster and add nodes in a short period of time.

### Anything else we need to know?
Ref: https://github.com/kubernetes/kubernetes/pull/105272/files

_No response_

### Kubernetes version

<details>
Cutting edge 1.30
```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>
EKS
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by mengqiy** [comment]
cc: @wojtek-t  since you wrote https://github.com/kubernetes/kubernetes/pull/105272/files

/sig node
/sig scalability

**Comment by sftim** [comment]
If the node used a random timer (eg uniform over 1m to 5m) for its first status update, and then stuck to _n_ second intervals with small jitter, that would go a long way towards smoothing out load in this scenario.

**Comment by sftim** [comment]
I also wonder if this is more of a feature request than a bug.

**Comment by AnishShah** [comment]
/remove-kind bug
/kind feature

Removing the bug label as this is a feature request. We welcome any changes for this issue.

**Comment by dims** [comment]
@mengqiy trying for a fix in 1.32? (reminder code freeze is very close!)

**Comment by mengqiy** [comment]
Created PR https://github.com/kubernetes/kubernetes/pull/128394 and hopefully we can get it merged before 1.32 code freeze.

**Comment by liggitt** [comment]
reopening, https://github.com/kubernetes/kubernetes/pull/128394 was rolled back because it introduced flakes
